### PR TITLE
feat(escalating-issues): Add archived analytics to issue stream

### PR DESCRIPTION
**Pull request by jangjodi**
part of https://github.com/getsentry/sentry/issues/49508

### CODE REVIEW

File: static/app/views/issueList/actions/index.spec.jsx [hunk 1]
```diff
@@ -233,10 +235,57 @@ describe('IssueListActions', function () {
...
+      expect.anything(),
+      expect.objectContaining({
+        query: {
+          id: ['1'],
+          project: [1],
+        },
+        data: {
+          status: 'ignored',
+          statusDetails: {},
+          substatus: 'until_escalating',
```
Comment (scttcper) [inline static/app/views/issueList/actions/index.spec.jsx:275]: can you update your branch i changed this string, sorry

Comment (scttcper) [reply static/app/views/issueList/actions/index.spec.jsx:275]: should be archived_until_escalating

File: static/app/utils/analytics/issueAnalyticsEvents.tsx [hunk 2]
```diff
@@ -136,6 +136,7 @@ export type IssueEventParameters = {
     search_source: string;
     search_type: string;
   };
+  'issues_stream.archived': {status_details?: string; substatus?: string};
```
[the remark this answers is not in the document]
Comment (jangjodi) [reply static/app/utils/analytics/issueAnalyticsEvents.tsx:139]: we're just looking for clicks on archive

File: static/app/views/issueList/actions/index.tsx [hunk 3]
```diff
@@ -136,6 +137,14 @@ function IssueListActions({
       'issue-list-removal-action'
     );
 
+    if (data.status === 'ignored') {
```
[the remark this answers is not in the document]
Comment (jangjodi) [reply static/app/views/issueList/actions/index.tsx:140]: @NisanthanNanthakumar in this case, we want analytics for any type of archive (ie. archive forever, archive until it happens once from now, archive until escalating etc.) so I didn't put it behind the flag

File: static/app/views/issueList/actions/index.spec.jsx (hunk 1 above)
Comment (jangjodi) [reply static/app/views/issueList/actions/index.spec.jsx]: nice, thanks for letting me know!

**Review by scttcper — APPROVED** [review]

**Pull request merged**